pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -176,19 +178,33 @@ func readInstanceID(cfg *VSphereConfig) (string, error) {
 		}
 	}
 	if svm == nil {
-		return "", fmt.Errorf("unable to retrieve vm reference from vSphere")
+		return "", "", fmt.Errorf("unable to retrieve vm reference from vSphere")
 	}
 
 	var vm mo.VirtualMachine
-	err = s.Properties(ctx, svm.Reference(), []string{"name"}, &vm)
+	err = s.Properties(ctx, svm.Reference(), []string{"name", "resourcePool"}, &vm)
 	if err != nil {
-		return "", err
+		return "", "", err
+	}
+
+	var cluster string
+	if vm.ResourcePool != nil {
+		// Extract the Cluster Name if VM belongs to a ResourcePool
+		var rp mo.ResourcePool
+		err = s.Properties(ctx, *vm.ResourcePool, []string{"parent"}, &rp)
```

Comment:
We may want to glog the errors in this block.  The errors may typically be meaningless since we don't always expect the vm to be in a cluster, but its possible for a more meaningful error to occur here.